File: [34]在排序数组中查找元素的第一个和最后一个位置.cpp
#include <vector>
using namespace std;

class Solution {
public:
	vector<int> searchRange(vector<int>& nums, int target) {
		if (nums.empty())
			return vector<int> {-1, -1};

		vector<int> ans(2);
		int left = 0;
		int right = nums.size();

		while (left < right) {
			int mid = (left + right) / 2;
			if (nums[mid] >= target)
				right = mid;
			else
				left = mid + 1;
		}
		if (left == nums.size() || nums[left] != target) {
			ans[0] = -1;
			ans[1] = -1;
			return ans;
		}
		else {
			ans[0] = right;

			left = right;
			right = nums.size();
			while (left < right) {
				int mid = (left + right) / 2;
				if (nums[mid] <= target)
					left = mid + 1;
				else
					right = mid;
			}
		}
		ans[1] = left - 1;

		return ans;
	}
};

File: solutions/[34]在排序数组中查找元素的第一个和最后一个位置.py
from typing import List


class Solution:
    def searchRange(self, nums: List[int], target: int) -> List[int]:
        l_p, r_p = -1, -1
        l, r = 0, len(nums)
        while l < r:
            mid = (l+r) // 2
            if nums[mid] < target:
                l = mid + 1
            else:
                r = mid
        if l < len(nums) and nums[l] == target:
            l_p = l
            l, r = 0, len(nums)
            while l < r:
                mid = (l+r) // 2
                if nums[mid] > target:
                    r = mid
                else:
                    l = mid + 1
            r_p = l-1

        return l_p, r_p